tests/test_focusAnalysis.py

```diff
@@ -0,0 +1,117 @@
+
+from lsst.summit.extras import SpectralFocusAnalyzer, NonSpectralFocusAnalyzer  # noqa: E402
+import lsst.summit.utils.butlerUtils as butlerUtils  # noqa: E402
+
+
+class FocusAnalysisTestCase(lsst.utils.tests.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.butler = butlerUtils.makeDefaultLatissButler()
+        except FileNotFoundError:
+            raise unittest.SkipTest("Skipping tests that require the LATISS butler repo.")
+
+        # dataIds have been chosen match those of a spectral focus sweep which
+        # is in the LATISS-test-data collection at the summit, TTS and USDF
+        cls.dayObs = 20220628
+        cls.seqNums = range(280, 288+1)
+        cls.focusAnalyzer = SpectralFocusAnalyzer()
+
+        # default is 3 boxes, so setting four tests the generality of the code.
+        # The values have been picked to land on the pretty small spectrum we
+        # have in the test data.
+        cls.focusAnalyzer.setSpectrumBoxOffsets([1600, 1700, 1800, 1900])
+
+    def test_run(self):
+        # we don't check the plots, but set doDisplay to True to check the
+        # plots are generated without error
+        self.focusAnalyzer.getFocusData(self.dayObs, self.seqNums, doDisplay=True)
+        result = self.focusAnalyzer.fitDataAndPlot()
+        self.assertIsInstance(result, Iterable)
+        self.assertEqual(len(result), len(self.focusAnalyzer.getSpectrumBoxOffsets()))
+
+        for number in result:
+            # check they're all numbers, non-nan, and vaguely sensible
+            self.assertGreater(number, -1.0)
+            self.assertLess(number, 1.0)
+
+
+class NonSpectralFocusAnalysisTestCase(lsst.utils.tests.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.butler = butlerUtils.makeDefaultLatissButler()
+        except FileNotFoundError:
+            raise unittest.SkipTest("Skipping tests that require the LATISS butler repo.")
+
+        # dataIds have been chosen match those of a non-spectral focus sweep
+        # which is in the LATISS-test-data collection at the summit, TTS and
+        # USDF
+        cls.dayObs = 20220405
+        cls.seqNums = range(523, 531+1)
+        cls.focusAnalyzer = NonSpectralFocusAnalyzer()
+
+    def test_run(self):
+        # we don't check the plots, but set doDisplay to True to check the
+        # plots are generated without error
+        self.focusAnalyzer.getFocusData(self.dayObs, self.seqNums, doDisplay=True)
+        result = self.focusAnalyzer.fitDataAndPlot()
+        self.assertIsInstance(result, dict)
+
+        # result is a dict which looks like this:
+        # {'fwhmFitMin': 0.029221417454391708,
+        #  'ee90FitMin': 0.0754762884665358,
+        #  'ee80FitMin': 0.07188778363981346,
+        #  'ee50FitMin': 0.13998855716378267}
+        self.assertEqual(len(result), 4)
+
+        for k, v in result.items():
+            # check they're all numbers, non-nan, and vaguely sensible
```

Comment:
Can you expand this comment with a brief explanation why the values must all be between 0 and 1?

Reply:
Yup, I've done the same here, thanks.